src/libpmemset/file.c

```diff
@@ -0,0 +1,145 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2020, Intel Corporation */
+
+/*
+ * file.c -- implementation of common file API
+ */
+
+#include <libpmem2.h>
+
+#include "alloc.h"
+#include "file.h"
+#include "pmemset_utils.h"
+
+struct pmemset_file {
+	bool close;
+	struct pmem2_source *pmem2_src;
+#ifdef _WIN32
+		HANDLE handle;
+#else
+		int fd;
+#endif
+};
+
+/*
+ * pmemset_file_get_pmem2_source -- retrieves the pmem2_source from pmemset_file
+ */
+struct pmem2_source *
+pmemset_file_get_pmem2_source(struct pmemset_file *file)
+{
+	return file->pmem2_src;
+}
+
+/*
+ * pmemset_file_init -- initializes pmemset_file structure based on the values
+ *                      stored in pmem2_source
+ */
+static int
+pmemset_file_init(struct pmemset_file *file, struct pmem2_source *pmem2_src)
+{
+	int ret;
+#ifdef _WIN32
+	HANDLE handle;
+	ret = pmem2_source_get_handle(pmem2_src, &handle);
+	file->handle = handle;
+#else
+	int fd;
+	ret = pmem2_source_get_fd(pmem2_src, &fd);
+	file->fd = fd;
+#endif
+	if (ret)
+		return ret;
+
+	file->pmem2_src = pmem2_src;
+
+	return 0;
+}
+
+/*
+ * pmemset_file_from_file -- allocates pmemset_file structure and initializes it
+ *                           based on the provided path to the file
+ */
+int
+pmemset_file_from_file(struct pmemset_file **file, char *path,
+		struct pmemset_config *cfg)
```

Comment:
Is *cfg* used anywhere?

Reply:
create file disposition will be taken from cfg (in pmemset_file_create_xxx) - not in the scope of this PR